internal/cortex/provider.go

```diff
@@ -36,14 +36,14 @@ func Provider() *schema.Provider {
 			},
 			"api_key": {
 				Type:        schema.TypeString,
-				Optional:    true,
+				Required:    true,
```

Comment:
This is optional because it can be set at the resource level. This is so one provider can configure multiple tenants.

Reply:
Yes, but if the value is not defined both at the provider level and resource level, I can still proceed with terraform plan and it only gives an error during apply which is kind of looks weird. So better to make this **required** at the provider level and then it can be overridden at the resource level if required.

Reply:
This is the comment for tenant_id.

Tenant_id is not required by cortex, if your have not enabled auth, then you do not need it.

Reply:
got it